options.go

```diff
@@ -1,17 +1,48 @@
-package hibpsync
+package hibp
 
 import (
 	"context"
 	"io"
 )
 
+// ProgressFunc represents a type of function that can be used to report progress of a sync operation.
+// The parameters are as follows:
+// - lowest: The lowest prefix that has been processed so far (due to concurrent operations, there is a window of
+// prefixes that are possibly being processed at the same time, "lowest" refers to the range with the lowest prefix).
+// - current: The current prefix that is being processed, i.e. for which the ProgressFunc gets invoked.
+// - to: The highest prefix that will be processed.
+// - processed: The number of prefixes that have been processed so far.
+// - remaining: The number of prefixes that are remaining to be processed.
+// The function should return an error if the operation should be aborted.
+type ProgressFunc func(lowest, current, to, processed, remaining int64) error
+
 type commonConfig struct {
 	dataDir       string
 	noCompression bool
 }
 
+type CommonOption func(config *commonConfig)
+
+// WithDataDir sets the data directory for all operations.
+// The directory will be created it if it does not exist.
+// Default: "./.hibp-data"
+func WithDataDir(dataDir string) CommonOption {
+	return func(c *commonConfig) {
+		c.dataDir = dataDir
+	}
+}
+
+// WithNoCompression disables compression when writing/reading the file-based database.
+// When the local dataset exists already, this can only be used if the dataset has been created with the same setting.
+// This seriously increases the amount of storage required.
+// Default: false
+func WithNoCompression() CommonOption {
```

Comment:
nit
```suggestion
// WithoutCompression disables compression when writing/reading the file-based database.
// When the local dataset exists already, this can only be used if the dataset has been created with the same setting.
// This seriously increases the amount of storage required.
// Default: false
func WithoutCompression() CommonOption {
```

Reply:
Done 👍